ultradns-rest/src/main/java/denominator/ultradns/UltraDNSRestProvider.java

```diff
@@ -0,0 +1,187 @@
+package denominator.ultradns;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.inject.Singleton;
+
+import dagger.Provides;
+import denominator.BasicProvider;
+import denominator.CheckConnection;
+import denominator.DNSApiManager;
+import denominator.QualifiedResourceRecordSetApi;
+import denominator.ResourceRecordSetApi;
+import denominator.ZoneApi;
+import denominator.config.ConcatBasicAndQualifiedResourceRecordSets;
+import denominator.config.NothingToClose;
+import denominator.config.WeightedUnsupported;
+import denominator.profile.GeoResourceRecordSetApi;
+//import UltraDNSRestErrorDecoder.UltraDNSError;
+import feign.Feign;
+import feign.Logger;
+import feign.Request.Options;
+import feign.form.FormEncoder;
+import feign.gson.GsonDecoder;
+import feign.gson.GsonEncoder;
+
+import static dagger.Provides.Type.SET;
+
+public class UltraDNSRestProvider extends BasicProvider {
+
+  private final String url;
+
+  public UltraDNSRestProvider() {
```

Comment:
Do we need this default constructor for any specific reason? I would suggest to have only the parameterized constructor, unless the default one is required for some framework needs.